src/java.base/share/classes/java/util/UUID.java

```diff
@@ -181,6 +181,70 @@ public static UUID nameUUIDFromBytes(byte[] name) {
         return new UUID(md5Bytes);
     }
 
+    /**
+     * Static factory to create a version 7 (time-based) {@code UUID} with the current
+     * Unix timestamp in milliseconds.
+     *
+     * The {@code UUID} embeds the current Unix Epoch timestamp in milliseconds using
+     * {@link System#currentTimeMillis()} into the first 6 bytes, sets the version and
+     * variant bits as per the specification and fills the remaining bytes with random
+     * data from a cryptographically strong pseudo-random number generator.
+     *
+     * @return a {@code UUID} generated with the current Unix timestamp
+     *
+     * @spec RFC 9562
+     */
+    public static UUID unixEpochTimeMillis() {
+        long timestamp = System.currentTimeMillis();
+        return unixEpochTimeMillis(timestamp);
+    }
+
+    /**
+     * Static factory to create a version 7 (time-based) {@code UUID} with a user-supplied
+     * Unix timestamp in milliseconds.
+     *
+     * The {@code UUID} embeds the provided Unix Epoch timestamp in milliseconds into
+     * the first 6 bytes, sets the version and variant bits as per the specification,
+     * and fills the remaining bytes with random data from a cryptographically strong
+     * pseudo-random number generator.
+     *
+     * <p><strong>Note:</strong> The timestamp must be a Unix Epoch timestamp in milliseconds in order
+     * to be compliant with <a href="https://datatracker.ietf.org/doc/html/rfc9562">RFC 9562</a>.
+     *
+     * @param timestamp
+     *        A Unix epoch timestamp in milliseconds which must fit in to 48 bits
+     *
+     * @return a {@code UUID} generated using the provided timestamp
+     *
+     * @throws IllegalArgumentException if the timestamp is negative or exceeds 48 bits
+     *
+     * @spec RFC 9562
+     */
+    public static UUID unixEpochTimeMillis(long timestamp) {
+        if (timestamp < 0 || (timestamp >>> 48) != 0) {
+            throw new IllegalArgumentException("Timestamp must be a 48-bit Unix Epoch time in milliseconds.");
+        }
```

Comment:
Sadly, the RFC does not make any provision for times before 1/1/1970, so there is an incomplete mapping between epochTime in other contexts that allows negative values to represent earlier times.
The test for positive 48 bits could be as simple as:
```suggestion
        if (timestamp >> 48) != 0) {
            throw new IllegalArgumentException("Timestamp must be an unsigned 48-bit Unix Epoch time in milliseconds.");
        }
```